STM32Cube/Tasks/flight_phase.c

```diff
@@ -8,53 +8,113 @@
 #include "flight_phase.h"
 #include "millis.h"
 #include "printf.h"
+#include "log.h"
+#include "otits.h"
 
 #define BOOST_TIME_MS 10000 //time from inj valve open to motor bunout
 #define RECOVERY_DELAY_MS 15000 //time from inj valve open to recovery deployment
 
-enum FLIGHT_PHASE flight_state = PHASE_PRELAUNCH;
-float injOpenTime;
-float deploymentTime;
+static FLIGHT_PHASE flight_state = PHASE_PRELAUNCH;
+static float injOpenTime;
+static float deploymentTime;
 
 EventGroupHandle_t flightPhaseEventsHandle = NULL;
 #define INJ_OPEN (xEventGroupGetBits(flightPhaseEventsHandle) & INJ_OPEN_BIT)
 #define RECOVERY_DEPLOYMENT (xEventGroupGetBits(flightPhaseEventsHandle) & RECOVERY_DEPLOYMENT_BIT)
 
+static Otits_Result_t test_flightPhase() {
+    Otits_Result_t res;
+    float time = millis_();
 
-void flightPhaseInit()
-{
+    // check that millis matches tick count to a reasonable degree TODO: is this how timer works lmao
+    if (time - xTaskGetTickCount() > 2) {
+        res.info = "millis() value err";
+        res.outcome = TEST_OUTCOME_FAILED;
+        return res;
+    }
+
+    // must be in prelaunch if inj hasn't opened yet
+    if (!INJ_OPEN) {
+        if (flight_state != PHASE_PRELAUNCH) {
+            res.info = "prelaunch not detected";
+            res.outcome = TEST_OUTCOME_FAILED;
+            return res;
+        } else if (time > 30000) { // check inj_open hasn't gotten stuck TODO: what should this number be ??
```

Comment:
the thought with this is to check if CAN received an inj_open msg and set the bit correctly. is this worth doing? is there a max time thats actually reasonable to test with?

Reply:
I don't think there is a reasonable limit. The board will probably be on for an hour or two before the injector valve opens